content/posts/2022/kerberos-all-steps-workflow.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "Easily enable Kerberos for all the steps of a workflow"
+date: 2022-11-30T14:30:00+01:00
+tags:
+  - kerberos
+---
+
+Up until now, it was only possible to enable Kerberos support for a workflow on
+a step-by-step basis. This changes with REANA 0.9.0, which makes it easy to
+enable Kerberos authentication for an entire workflow by using only a single
+hint in its specification file.
+
+<!--more-->
+
+As an example, let's consider a workflow that uses Kerberos to access a folder
+on an external restricted storage backend like EOS to publish results there.
+This is how the `reana.yaml` specification file might look like:
+
+```yaml {hl_lines=[10]}
+workflow:
+  type: serial
+  specification:
+    steps:
+      - name: myfirststep
+        #...
+      - name: mysecondstep
+        #...
+      - name: publish
+        kerberos: true
+        environment: "reanahub/krb5"
+        commands:
+          - mkdir -p /eos/home/j/johndoe/myanalysis-outputs
+          - cp myplots/*.png /eos/home/j/johndoe/myanalysis-outputs/
+```
+
+Setting `kerberos: true` on each step that needs it works well if just a few
```

Comment:
You can specify `kerberos: true` analogously for each step of the workflow where Kerberos authentication is needed. This will work without problems, although it may be a bit tedious.

However, there are cases where Kerberos authentication is required even for orchestrating the workflow execution itself, for example when using the Snakemake workflow engine with input data objects living in a restricted data storage. Here, specifying `kerberos: true` for workflow steps only wouldn't suffice.

This is now possible with REANA 0.9.0 which makes it easy to enable Kerberos authentication for an entire workflow by using only a single hint in the workflow specification file.

Reply:
I have also divided the first part of the blog post in two sections:
- Enable Kerberos for a single workflow step
- Enable Kerberos at the workflow orchestration level